packages/relay/src/routes/post.ts

```diff
@@ -20,167 +15,54 @@ export default (
-            return
-        }
-
-        // check if state tree exists in current epoch
-        const isStateTreeValid = await synchronizer.stateTreeRootExists(
-            epochKeyProof.stateTreeRoot,
-            Number(epochKeyProof.epoch),
-            epochKeyProof.attesterId
-        )
-        if (!isStateTreeValid) {
-            res.status(400).json({ error: 'Invalid State Tree' })
-            return
-        }
-
-        // check if proof is valid
-        const isProofValid = await epochKeyProof.verify()
-        if (!isProofValid) {
-            res.status(400).json({ error: 'Invalid proof' })
-            return
-        }
-
-        const appContract = new ethers.Contract(APP_ADDRESS, ABI)
-
-        // post content
-        let calldata: any
-        let cid: any
-        if (content) {
-            // if the content is not empty, post the content
-            calldata = appContract.interface.encodeFunctionData('post', [
-                epochKeyProof.publicSignals,
-                epochKeyProof.proof,
-                content,
-            ])
-
-            // store content into helia ipfs node with json plain
-            const { json } = await eval("import('@helia/json')")
-            const heliaJson = json(helia)
-            const IPFSContent = {
-                content: content,
+            const id = req.params.id
+            if (!id) {
+                console.log('id is undefined')
+                res.status(400).json({ error: 'id is undefined' })
             }
-            let file = await heliaJson.add(JSON.stringify(IPFSContent))
-            cid = file.toString()
-        }
-
-        const hash = await TransactionManager.queueTransaction(
-            APP_ADDRESS,
-            calldata
-        )
-
-        const epochKey = epochKeyProof.epochKey.toString()
 
-        // after post data stored in DB, should add 1 to epoch key counter
-        await addActionCount(db, epochKey, epoch, (txDB) => {
-            txDB.create('Post', {
-                content: content,
-                cid: cid,
-                epochKey: epochKey,
-                epoch: epoch,
-                transactionHash: hash,
-                status: 0,
-            })
-
-            return 1
-        })
-
-        res.json({
-            transaction: hash,
-        })
-    } catch (error: any) {
-        console.log(error)
-        res.status(500).json({ error })
-    }
-}
-
-async function fetchSinglePost(req, res, db: DB) {
-    try {
-        const id = req.params.id
-        if (!id) {
-            res.status(400).json({ error: 'id is undefined' })
-        }
-
-        const post = await db.findOne('Post', {
-            where: {
-                _id: id,
-            },
+            const post = await postService.fetchSinglePost(id, db, undefined)
+            if (!post) {
+                console.log(`post is not found: ${id}`)
+                res.status(404).json({ error: `post is not found: ${id}` })
```

Comment:
should add return to finish this request
or should throw error like comment api

Reply:
I think this one is find, since it is the end of method